CMake/Settings.cmake

```diff
@@ -25,15 +25,6 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
   )
 endif ()
 
-if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
Have to remove this because it causes error for clang-tidy

Reply:
Hm, that's annoying 😢 Maybe we can keep them if `lint` is not enabled?

Reply:
Curious what the error is ?